gneiss/plot/_dendrogram.py

```diff
@@ -0,0 +1,235 @@
+                name of right child node
+        """
+        self.rescale(width, height)
+        result = {}
+        for node in self.postorder(include_self=True):
+            children = {'child%d' % i: n.name
+                        for i, n in enumerate(node.children)}
+            coords = {'x': node.x2, 'y': node.y2}
+            is_tip = {'is_tip': node.is_tip()}
+            result[node.name] = {**coords, **children, **is_tip}
+        result = pd.DataFrame(result).T
+
+        # reorder so that x and y are first
+        cols = list(result)
+        cols.insert(0, cols.pop(cols.index('y')))
+        cols.insert(0, cols.pop(cols.index('x')))
+        result = result.ix[:, cols]
+        return result
+
+    @abc.abstractmethod
+    def rescale(self, width, height):
+        pass
+
+
+class UnrootedDendrogram(Dendrogram):
+    aspect_distorts_lengths = True
+
+    def __init__(self, **kwargs):
+        """ Constructs a UnrootedDendrogram object for visualization.
+
+        Parameters
+        ----------
+        use_lengths: bool
+            Specifies if the branch lengths should be included in the
+            resulting visualization (default True).
+        """
+        super().__init__(**kwargs)
+
+    @classmethod
+    def from_tree(cls, tree):
+        """ Creates an UnrootedDendrogram object from a skbio tree.
+
+        Parameters
+        ----------
+        tree : skbio.TreeNode
+            Input skbio tree
+
+        Returns
+        -------
+        UnrootedDendrogram
+        """
+        for n in tree.postorder(include_self=True):
+            n.__class__ = UnrootedDendrogram
+        tree._cache_ntips()
+        return tree
+
+    def rescale(self, width, height):
+        """ Find best scaling factor for fitting the tree in the canvas
+
+        Parameters
+        ----------
+        width : float
+            width of the canvas
+        height : float
+            height of the canvas
+
+        Returns
+        -------
+        best_scaling : float
+            largest scaling factor in which the tree can fit in the canvas.
+
+        Notes
+        -----
+        This function has a little bit of recursion.  This will
+        need to be refactored to remove the recursion.
+        """
+
+        angle = 2*numpy.pi / self._n_tips
+        # this loop is a horrible brute force hack
+        # there are better (but complex) ways to find
+        # the best rotation of the tree to fit the display.
+        best_scale = 0
+        for i in range(60):
+            direction = i / 60.0 * numpy.pi
+            points = self.update_coordinates(1.0, 0, 0, direction, angle)
+            xs = [x for (x, y) in points]
+            ys = [y for (x, y) in points]
+
+            # double check that the tree fits within the margins
+            scale = min(float(width) / (max(xs) - min(xs)),
+                        float(height) / (max(ys) - min(ys)))
+            scale *= 0.95  # extra margin for labels
```

Comment:
How was the factor 0.95 decided? Will the labels always occupy an absolute amount of space (e.g., 12 px) or relative (e.g., 1 em)?